controllers/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -294,6 +297,18 @@ func (r *PersistentVolumeClaimReconciler) determineScheduleAndRequeue(
 		return "", ErrConnNotFoundRequeueNeeded
 	}
 
+	// check for storageclass schedule annotation.
+	sc := &storagev1.StorageClass{}
+	err = r.Client.Get(ctx, types.NamespacedName{Name: *pvc.Spec.StorageClassName}, sc)
```

Comment:
Will this work for the case where storageclass is not specified in the PVC but there is a default storagecluster in the cluster.

Reply:
> Will this work for the case where storageclass is not specified in the PVC but there is a default storagecluster in the cluster.

yes, in that case default SC name will be added to `pvc.Spec` by k8s.

```bash
pm@dhcp53-176:~/github.com/rook/rook/deploy/examples/csi/rbd$ k get sc
NAME                        PROVISIONER                  RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
rook-ceph-block (default)   rook-ceph.rbd.csi.ceph.com   Delete          Immediate           false                  9m54s
standard                    k8s.io/minikube-hostpath     Delete          Immediate           false                  15m
pm@dhcp53-176:~/github.com/rook/rook/deploy/examples/csi/rbd$ k create -f pvc.yaml
persistentvolumeclaim/rbd-pvc-new created
pm@dhcp53-176:~/github.com/rook/rook/deploy/examples/csi/rbd$ k get pvc
NAME          STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
rbd-pvc-new   Bound    pvc-d9db8f4e-beb7-4760-9aa7-1c89d9e6bfb1   1Gi        RWO            rook-ceph-block   3s
pm@dhcp53-176:~/github.com/rook/rook/deploy/examples/csi/rbd$ k get pvc rbd-pvc-new -oyaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-provisioner: rook-ceph.rbd.csi.ceph.com
    volume.kubernetes.io/storage-provisioner: rook-ceph.rbd.csi.ceph.com
  creationTimestamp: "2024-06-19T08:06:18Z"
  finalizers:
  - kubernetes.io/pvc-protection
  name: rbd-pvc-new
  namespace: rook-ceph
  resourceVersion: "3013"
  uid: d9db8f4e-beb7-4760-9aa7-1c89d9e6bfb1
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
  storageClassName: rook-ceph-block
  volumeMode: Block
  volumeName: pvc-d9db8f4e-beb7-4760-9aa7-1c89d9e6bfb1
  ```

Reply:
can you please check the pvc.yaml did you set the storageclass while you are creating the PVC? or it got added by kubernetes itself?

Reply:
yes, I did remove while create 
```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: rbd-pvc-new
spec:
  accessModes:
    - ReadWriteOnce
  volumeMode: Block
  resources:
    requests:
      storage: 1Gi
 ```